web/daemon/server.go

```diff
@@ -668,6 +672,60 @@ func GetStatus(c *gin.Context) {
 	}
 }
 
+// @Summary Archive file(s)
+// @Description Archives file(s) with the
+// @Accept json
+// @Success 204 {object} response.Empty "If file(s) was archived"
+// @Failure 400 {object} response.Error
+// @Failure 403 {object} response.Empty
+// @Failure 404 {object} response.Empty
+// @Failure 500 {object} response.Error
+// @Param id path string true "Server Identifier"
+// @Param filename path string true "Destination"
+// @Router /daemon/server/{id}/archive/{filename} [post]
+func Archive(c *gin.Context) {
+	item, _ := c.Get("server")
+	server := item.(*programs.Program)
+	var files []string
+	if err := c.BindJSON(&files); response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+	destination := c.Param("filename")
+
+	err := server.ArchiveItems(files, destination)
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+	} else {
+		c.Status(http.StatusNoContent)
+	}
+}
+
+// @Summary Extract files
+// @Description Extracts files from an archive
+// @Accept json
+// @Produce json
+// @Success 204 {object} response.Empty "If file was extracted"
+// @Failure 400 {object} response.Error
+// @Failure 403 {object} response.Empty
+// @Failure 404 {object} response.Empty
+// @Failure 500 {object} response.Error
+// @Param id path string true "Server Identifier"
+// @Param filename path string true "File name"
+// @Param destination path string true "Destination directory (URI Parameter)"
+// @Router /daemon/server/{id}/extract/{filename} [get]
+func Extract(c *gin.Context) {
+	item, _ := c.Get("server")
+	server := item.(*programs.Program)
+
+	targetPath := c.Param("filename")
+	destination := c.Query("destination")
```

Comment:
No checks are done to ensure destination isn't empty, that is a client error and should be reported as such